packages/core/src/application.ts

```diff
@@ -77,12 +90,13 @@ export class Application extends Context {
     ctor: Constructor<T>,
     name?: string,
   ): Binding<T> {
+    debug('Adding server %s', name || ctor.name);
     const binding = createBindingFromClass(ctor, {
       name,
       namespace: CoreBindings.SERVERS,
       type: CoreTags.SERVER,
       defaultScope: BindingScope.SINGLETON,
-    });
+    }).apply(asLifeCycleObserver);
```

Comment:
A typical LifeCycleObserver is bound with the following metadata

```ts
    const binding = createBindingFromClass(ctor, {
	      name,
	      namespace: CoreBindings.LIFE_CYCLE_OBSERVERS,
	      type: CoreTags.LIFE_CYCLE_OBSERVER,
	      defaultScope: BindingScope.SINGLETON,
	    }).tag(CoreTags.LIFE_CYCLE_OBSERVER);
```

When a server is registered as an observer, some of the metadata will be different:

- namespace: `SERVERS` vs. `LIFE_CYCLE_OBSERVERS`
- type: `SERVER` vs. `LIFE_CYCLE_OBSERVER`

Is it intended and acceptable? I am concerned about possible confusion it may create.

Reply:
The only requirement is to be tagged as `CoreTags.LIFE_CYCLE_OBSERVER`. I have updated code accordingly.